plugins/commando.c

```diff
@@ -1124,17 +1149,16 @@ static struct command_result *json_commando_blacklist(struct command *cmd,
 			continue;
 		}
 		// old list overlaps combined into the entry we are adding
-		nextmerged = blacklist_merge(entry, &blacklist[i], &blacklist[i+1]);
-		tal_arr_expand(&newblacklist, blacklist[i]);
-		entry = NULL;
-		if (nextmerged) { i = i + 1; }
+		blacklist_merge(entry, &blacklist[i]);
```

Comment:
OK, merge these fixes into the previous commit, and have this one just do what it promises: save blacklist into datastore.